pkg/server/operands_handler.go

```diff
@@ -0,0 +1,73 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/openshift/console/pkg/serverutils"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+	"k8s.io/klog"
+)
+
+type OperatorMeta struct {
+	Name      string
+	Namespace string
+}
+
+func operandsListHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+	operatorMeta := parseOperatorNameAndNamespace(r.URL.Path)
+
+	cfg := &action.Configuration{}
+	err := cfg.Load()
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to set load operator client: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+	}
+	// We need to manually set the namespace here since the kubectl-operator
+	// is setting the namespace to "default" upon calling the `cfg.Load()` method
+	if len(operatorMeta.Namespace) != 0 {
+		cfg.Namespace = operatorMeta.Namespace
+	}
+
+	operatorListOperands := action.NewOperatorListOperands(cfg)
+
+	operandsList, err := operatorListOperands.Run(context.TODO(), operatorMeta.Name)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to list operands: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	resp, err := json.Marshal(operandsList)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to marshal the list operands response: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+	}
+	w.Write(resp)
+}
+
+func parseOperatorNameAndNamespace(urlPath string) *OperatorMeta {
+	nameAndNamespace := strings.SplitN(urlPath, ".", 2)
```

Comment:
Can you share what the URL will look like? The name/namespace is split on a `.` character?

We could consider just making these values query parameters if it's easier to parse.

Reply:
the format is currently:
`<cluster-url>/api/list-operands/<operator-name>.<operator-namespace>`

the `<operator-name>.<operator-namespace>` is basically format what the `kubectl get operators` returns, since all operators are installed in a particualr NS, even if cluster operator.
eg.
`<cluster-url>/api/list-operands/etcd.openshift-etcd`

Reply:
> We could consider just making these values query parameters if it's easier to parse.

yeah was considering that as well, but not sure if there is any benefit. I could imagine that if we would want to list operands for multiple operators by single call. Is that something we should  think about ?

Reply:
I'm concerned a period in a namespace name would cause problems. (I'm not sure offhand if that's allowed.)

The other benefit is that we don't have to parse the path. We just read the query parameters. So it's simpler. Since this is an internal console API, what we use doesn't matter too much, so I would do the simplest thing.